contrib/inventory/rackhd.py

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/python
+import json
+import requests
+import os
+import argparse
+import types
+
+RACKHD_URL = 'http://localhost:8080'
+
+class RackhdInventory(object):
+    def __init__(self, nodeids):
+        self._inventory = {}
+        for nodeid in nodeids:
+            self._load_inventory_data(nodeid)
+        output = '{\n'
+        for nodeid,info in self._inventory.iteritems():
+            output += self._format_output(nodeid, info)
+            output += ',\n'
+        output = output[:-2]
+        output += '}\n'
+        print (output)
+
+    def _load_inventory_data(self, nodeid):
+        info = {}
+        info['ohai'] = RACKHD_URL + '/api/common/nodes/{0}/catalogs/ohai'.format(nodeid )
+        info['lookup'] = RACKHD_URL + '/api/common/lookups/?q={0}'.format(nodeid)
+
+        results = {}
+        for key,url in info.iteritems():
+            r = requests.get( url, verify=False)
+            results[key] = r.text
+
+        self._inventory[nodeid] = results
+
+    def _format_output(self, nodeid, info):
+        output = ''
+        try:
+            node_info = json.loads(info['lookup'])
+            ipaddress = ''
+            if len(node_info) > 0:
+                ipaddress = node_info[0]["ipAddress"]
+            output += '  "' + nodeid + '" : {\n'
+            output += '    "hosts": [ "' + ipaddress + '" ],\n'
+            output += '    "vars" : {\n'
+            for key,result in info.iteritems():
+                output += '      "' + key + '": ' + json.dumps(json.loads(result), sort_keys=True, indent=2) + ',\n'
+            output += '      "ansible_ssh_user": "renasar"\n'
+            output += '    }\n'
+            output += '  }\n'
+        except KeyError:
```

Comment:
I'm rather confused here. It looks like we are both using the `json` module to create JSON, as well as hand crafting JSON?

I strongly recommend, dropping this functionality, and building out an actual python datastrcuture, and just using `json.dumps` on the final result.